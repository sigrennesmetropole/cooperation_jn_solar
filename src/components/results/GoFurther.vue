<script setup lang="ts">
import PhotoVoltaique from '@/components/results/PhotoVoltaique.vue'
import LabelsProfitability from '@/components/results/LabelsProfitability.vue'
import qualiLabels from '@/assets/illustrations/quali_labels.svg'
import qualitenr from '@/assets/illustrations/qualitEnR.png'
import qualibat from '@/assets/illustrations/qualibat.avif'
import qualifelec from '@/assets/illustrations/qualifelec.png'
import profitability from '@/assets/illustrations/profitability.svg'

const props = defineProps<{
  isPdf: boolean
}>()
</script>

<template>
  <div
    class="flex flex-col gap-6 w-[100%] h-fit bg-white rounded-xl py-6 px-8 mx-auto shadow-md"
  >
    <h2 class="font-dm-sans font-bold text-2xl">
      Plus d'informations pour concrétiser votre projet
    </h2>
    <PhotoVoltaique :isPdf="props.isPdf"></PhotoVoltaique>
    <div class="flex flex-row gap-8">
      <LabelsProfitability
        link="https://www.photovoltaique.info/fr/preparer-un-projet/quelles-demarches-realiser/choisir-son-installateur/#signes_de_qualite"
        :isPdf="props.isPdf"
      >
        <template v-slot:img>
          <img :src="qualiLabels" />
        </template>
        <template v-slot:title>
          <h2 class="font-dm-sans text-2xl font-bold">
            3 labels de qualification pour vous guider dans le choix de votre
            installateur
          </h2>
        </template>
        <template v-slot:icon>
          <div
            class="py-[18px] px-6 flex flex-row justify-between w-[100%] bg-white rounded-xl"
          >
            <img :src="qualibat" alt="" class="h-10 w-10" />
            <img :src="qualitenr" alt="" class="h-10 w-[79px]" />
            <img :src="qualifelec" alt="" class="h-10 w-[34px]" />
          </div>
        </template>
      </LabelsProfitability>
      <LabelsProfitability
        link="https://www.photovoltaique.info/fr/preparer-un-projet/quelles-demarches-realiser/choisir-son-modele-economique/"
        :isPdf="props.isPdf"
      >
        <template v-slot:img>
          <img :src="profitability" />
        </template>
        <template v-slot:title>
          <h2 class="font-dm-sans text-2xl font-bold">
            Connaitre des ordres de grandeur de coûts et de rentabilité d'un
            projet photovoltaïque
          </h2>
        </template>
        <template v-slot:icon>
          <div class="h-full"></div>
        </template>
      </LabelsProfitability>
    </div>
  </div>
</template>
