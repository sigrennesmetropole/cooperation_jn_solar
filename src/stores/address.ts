import { ref } from 'vue'
import type { Ref } from 'vue'
import { defineStore } from 'pinia'

export const useAddressStore = defineStore('address', () => {
  const latitude: Ref<number> = ref(0)
  const longitude: Ref<number> = ref(0)
  const address: Ref<string> = ref('')
  const screenshotAddress: Ref<Blob | null> = ref(null)

  function setLatitudeAndLongitude(
    new_latitude: number,
    new_longitude: number
  ) {
    latitude.value = new_latitude
    longitude.value = new_longitude
  }

  function setAddress(new_address: string) {
    address.value = new_address
  }

  function resetLatitureAndLongitude() {
    setLatitudeAndLongitude(0, 0)
  }

  function resetAddress() {
    setAddress('')
  }

  return {
    latitude,
    longitude,
    setLatitudeAndLongitude,
    address,
    setAddress,
    resetAddress,
    resetLatitureAndLongitude,
    screenshotAddress,
  }
})
