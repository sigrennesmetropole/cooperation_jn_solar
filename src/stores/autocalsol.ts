import { ref } from 'vue'
import type { Ref } from 'vue'
import { defineStore } from 'pinia'
import type { AutocalsolResult } from '@/model/autocalsol.model'

export const useAutocalsolStore = defineStore('autocalsolStore', () => {
  const autocalsolResult: Ref<AutocalsolResult | null> = ref(null)

  function setAutocalsolResult(result: AutocalsolResult | null) {
    autocalsolResult.value = result
  }

  function resetAutocalsolResult() {
    autocalsolResult.value = null
  }

  return {
    autocalsolResult,
    setAutocalsolResult,
    resetAutocalsolResult,
  }
})
