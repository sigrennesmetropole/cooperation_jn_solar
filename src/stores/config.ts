import { ref } from 'vue'
import type { Ref } from 'vue'
import { defineStore } from 'pinia'
import type { ConfigType } from '@sigrennesmetropole/cooperation_jn_common_ui'

export const useConfigStore = defineStore('configStore', () => {
  const config: Ref<ConfigType | null> = ref(null)

  function setConfig(newConfig: ConfigType | null) {
    config.value = newConfig
  }

  return {
    config,
    setConfig,
  }
})
