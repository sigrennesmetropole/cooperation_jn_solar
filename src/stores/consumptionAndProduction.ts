import type { Ref } from 'vue'
import { ref } from 'vue'
import { defineStore } from 'pinia'

export const useConsumptionAndProductionStore = defineStore(
  'consumptionAndProduction',
  () => {
    const annualConsumption: Ref<number> = ref(0)

    function setAnnualConsumption(new_annual_consumption: number) {
      annualConsumption.value = new_annual_consumption
    }

    function resetAnnualConsumption() {
      setAnnualConsumption(0)
    }

    return {
      annualConsumption,
      setAnnualConsumption,
      resetAnnualConsumption,
    }
  }
)
