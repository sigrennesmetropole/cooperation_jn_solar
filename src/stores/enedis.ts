import { ref } from 'vue'
import type { Ref } from 'vue'
import { defineStore } from 'pinia'

export const useEnedisStore = defineStore('enedisStore', () => {
  const isEnedisRedirection: Ref<boolean> = ref(false)

  function setIsEnedisRedirection(newValue: boolean) {
    isEnedisRedirection.value = newValue
  }

  return {
    isEnedisRedirection,
    setIsEnedisRedirection,
  }
})
