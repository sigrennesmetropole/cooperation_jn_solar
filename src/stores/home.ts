import { ref } from 'vue'
import type { Ref } from 'vue'
import { defineStore } from 'pinia'

export const useHomeStore = defineStore('home', () => {
  const isTermOfUseAccepted: Ref<boolean> = ref(false)
  const displayError: Ref<boolean> = ref(false)

  function setIsTermOfUseAccepted(newValue: boolean) {
    isTermOfUseAccepted.value = newValue
  }

  function setDisplayError(newValue: boolean) {
    displayError.value = newValue
  }

  return {
    isTermOfUseAccepted,
    displayError,
    setIsTermOfUseAccepted,
    setDisplayError,
  }
})
