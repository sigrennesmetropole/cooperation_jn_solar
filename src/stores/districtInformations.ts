import { ref } from 'vue'
import type { Ref } from 'vue'
import { defineStore } from 'pinia'
import type { Viewpoint } from '@vcmap/core'
import type { Feature } from 'ol'
import { useInteractionsStore } from '@/stores/interactions'
import SelectDistrictInteraction from '@/interactions/selectDistrictInteractions'
import { RENNES_LAYER, useLayersStore } from '@/stores/layers'

export const useDistrictStore = defineStore('districtInformations', () => {
  const interactionsStore = useInteractionsStore()
  const layerStore = useLayersStore()

  const districtCode: Ref<number> = ref(0)
  const districtName: Ref<string> = ref('')
  const districtConsumption: Ref<number> = ref(0)
  const districtProduction: Ref<number> = ref(0)
  const districtNumberInstallations: Ref<number> = ref(0)
  const districtPointFeature: Ref<Feature | null> = ref(null)
  const previousViewPoint: Ref<Viewpoint | null> = ref(null)
  const newPointAbscissa: Ref<number> = ref(0)
  const newPointOrdinate: Ref<number> = ref(0)
  const checkboxChecked: Ref<boolean> = ref(false)
  const canBeDisplayed: Ref<boolean> = ref(true)
  function setDistrictIrisCode(newDistrictCode: number) {
    districtCode.value = newDistrictCode
  }

  function setDistrictName(newDistrictName: string) {
    districtName.value = newDistrictName
  }

  function setDistrictConsumption(newDistrictConsumption: number) {
    districtConsumption.value = newDistrictConsumption
  }

  function setDistrictProduction(newDistrictProduction: number) {
    districtProduction.value = newDistrictProduction
  }

  function setDistrictNumberInstallations(
    newDistrictNumberInstallations: number
  ) {
    districtNumberInstallations.value = newDistrictNumberInstallations
  }

  function setDistrictInformations(
    newDistrictCode: number,
    newDistrictName: string,
    newDistrictConsumption: number,
    newDistrictProduction: number,
    newDistrictNumberInstallations: number
  ) {
    setDistrictIrisCode(newDistrictCode)
    setDistrictName(newDistrictName)
    setDistrictConsumption(newDistrictConsumption)
    setDistrictProduction(newDistrictProduction)
    setDistrictNumberInstallations(newDistrictNumberInstallations)
  }

  function setNewPointFeatureOnSelectedDistrict(newPointFeature: Feature) {
    districtPointFeature.value = newPointFeature
  }

  function setNewCoordinates(newAbscissa: number, newOrdinate: number) {
    newPointAbscissa.value = newAbscissa
    newPointOrdinate.value = newOrdinate
  }

  function setCheckboxChecked(newAction: boolean) {
    checkboxChecked.value = newAction
    if (newAction && canBeDisplayed) {
      layerStore.enableLayer(RENNES_LAYER.iris)
      interactionsStore.enableInteraction(SelectDistrictInteraction)
    } else {
      layerStore.disableLayer(RENNES_LAYER.iris)
      interactionsStore.disableInteraction(SelectDistrictInteraction)
    }
  }

  function setCanBeDisplayed(isFarEnough: boolean) {
    canBeDisplayed.value = isFarEnough
    if (!isFarEnough) {
      setCheckboxChecked(false)
    }
  }

  function resetDistrictStore() {
    setDistrictInformations(0, '', 0, 0, 0)
  }

  return {
    districtCode,
    districtName,
    districtConsumption,
    districtProduction,
    districtNumberInstallations,
    districtPointFeature,
    previousViewPoint,
    newPointAbscissa,
    newPointOrdinate,
    checkboxChecked,
    canBeDisplayed,
    setDistrictIrisCode,
    setDistrictName,
    setDistrictConsumption,
    setDistrictProduction,
    setDistrictNumberInstallations,
    setDistrictInformations,
    setNewPointFeatureOnSelectedDistrict,
    setNewCoordinates,
    setCheckboxChecked,
    setCanBeDisplayed,
    resetDistrictStore,
  }
})
