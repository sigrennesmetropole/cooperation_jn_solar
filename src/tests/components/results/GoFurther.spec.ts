import { mount, VueWrapper } from '@vue/test-utils'
import GoFurther from '@/components/results/GoFurther.vue'
import { createTestingPinia } from '@pinia/testing'
import { useConfigStore } from '@/stores/config'
import configuration from '@/tests/config/configuration_test.json'

describe('LabelsProfitability', () => {
  let wrapper: VueWrapper
  let configStore

  beforeEach(async () => {
    const testingPinia = createTestingPinia()
    configStore = useConfigStore(testingPinia)
    configStore.config = configuration
    wrapper = mount(GoFurther, {
      global: {
        plugins: [testingPinia],
      },
    })
  })

  it('renders correctly', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })
  it('contains h2', () => {
    const GoFurtherTitle = wrapper.find('h2')
    expect(GoFurtherTitle.exists()).toBe(true)
    expect(GoFurtherTitle.classes()).toContain('text-2xl')
  })
})
