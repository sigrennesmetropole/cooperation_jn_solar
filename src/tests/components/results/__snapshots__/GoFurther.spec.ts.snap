// Vitest Snapshot v1

exports[`LabelsProfitability > renders correctly 1`] = `
"<div class=\\"flex flex-col gap-6 w-[100%] h-fit bg-white rounded-xl py-6 px-8 mx-auto shadow-md\\">
  <h2 class=\\"font-dm-sans font-bold text-2xl\\"> Plus d'informations pour concrétiser votre projet </h2>
  <div class=\\"flex flex-row gap-8 w-[100%] h-fit bg-white\\">
    <div class=\\"flex flex-row w-[50%] justify-center h-fit border border-neutral-300 rounded-lg px-5 py-8\\"><img src=\\"/src/assets/illustrations/photo_voltaique.svg\\"></div>
    <div class=\\"flex flex-col w-[50%] h-fit\\">
      <p class=\\"font-dm-sans text-base font-normal\\"> Le site de référence <span tabindex=\\"0\\" class=\\"inline-flex items-center cursor-pointer font-medium\\"><strong class=\\"underline\\"> photovoltaique.info</strong><img src=\\"/src/assets/icons/expand-small-bigger-retract-smaller-big.svg\\" class=\\"w-[14px] h-[14px] ml-1\\"></span> regroupe l’essentiel des informations techniques et réglementaires sur le photovoltaïque. </p><a href=\\"https://www.photovoltaique.info/fr/\\" target=\\"_blank\\" class=\\"bg-black shadow-sm rounded-lg gap-3 px-4 py-3 w-fit flex flex-row items-center justify-center mt-4\\"><img class=\\"w-5 h-5\\" src=\\"/src/assets/icons/icon-external-link-white-thin.svg\\" alt=\\"\\"><span class=\\"font-dm-sans text-white text-base font-bold\\"> Visiter le site </span></a>
    </div>
  </div>
  <div class=\\"flex flex-row gap-8\\">
    <div class=\\"flex flex-col gap-6 w-[100%] bg-slate-100 rounded-lg p-6 mx-auto\\"><img src=\\"/src/assets/illustrations/quali_labels.svg\\">
      <h2 class=\\"font-dm-sans text-2xl font-bold\\"> 3 labels de qualification pour vous guider dans le choix de votre installateur </h2>
      <div class=\\"py-[18px] px-6 flex flex-row justify-between w-[100%] bg-white rounded-xl\\"><img src=\\"/src/assets/illustrations/qualibat.avif\\" alt=\\"\\" class=\\"h-10 w-10\\"><img src=\\"/src/assets/illustrations/qualitEnR.png\\" alt=\\"\\" class=\\"h-10 w-[79px]\\"><img src=\\"/src/assets/illustrations/qualifelec.png\\" alt=\\"\\" class=\\"h-10 w-[34px]\\"></div>
      <div class=\\"flex flex-row cursor-pointer w-fit\\" tabindex=\\"0\\">
        <p class=\\"underline decoration-1 font-medium text-base\\"> Accéder à la page </p><img src=\\"/src/assets/icons/expand-small-bigger-retract-smaller-big.svg\\" class=\\"w-3.5 h-3.5 ml-2 mt-1\\">
      </div>
    </div>
    <div class=\\"flex flex-col gap-6 w-[100%] bg-slate-100 rounded-lg p-6 mx-auto\\"><img src=\\"/src/assets/illustrations/profitability.svg\\">
      <h2 class=\\"font-dm-sans text-2xl font-bold\\"> Connaitre des ordres de grandeur de coûts et de rentabilité d'un projet photovoltaïque </h2>
      <div class=\\"h-full\\"></div>
      <div class=\\"flex flex-row cursor-pointer w-fit\\" tabindex=\\"0\\">
        <p class=\\"underline decoration-1 font-medium text-base\\"> Accéder à la page </p><img src=\\"/src/assets/icons/expand-small-bigger-retract-smaller-big.svg\\" class=\\"w-3.5 h-3.5 ml-2 mt-1\\">
      </div>
    </div>
  </div>
</div>"
`;
